editor/src/components/canvas/canvas-strategies/strategies/convert-to-absolute-and-move-strategy.spec.browser2.tsx

```diff
@@ -1299,6 +1309,220 @@ describe('Convert to absolute/escape hatch', () => {
+
+      const canvas = initialEditor.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+      const viewCenter = canvasPoint({
+        x: viewBounds.left + viewBounds.width / 2,
+        y: viewBounds.top + viewBounds.height / 2,
+      })
+
+      await mouseDownAtPoint(canvas, viewCenter)
+      await mouseMoveToPoint(canvas, offsetPoint(viewCenter, canvasPoint({ x: 15, y: 15 })))
+
+      await initialEditor.dispatch(
+        [CanvasActions.setUsersPreferredStrategy(ConvertToAbsoluteAndMoveStrategyID)],
+        true,
+      )
+
+      await mouseUpAtPoint(canvas, offsetPoint(viewCenter, canvasPoint({ x: 15, y: 15 })))
+
+      expect(getPrintedUiJsCode(initialEditor.getEditorState())).toEqual(
+        makeTestProjectCodeWithSnippet(
+          `<View style={{ ...(props.style || {}) }} data-uid='aaa'>
+          <View
+            style={{ backgroundColor: '#aaaaaa33', width: 250, height: 300, position: 'absolute', left: 15, top: 15  }}
+            data-uid='bbb'
+            data-testid='bbb'
+          />
+        </View>`,
+        ),
+      )
+    })
+
+    it('works on a flow element with all pins', async () => {
+      const initialEditor = await renderTestEditorWithCode(
+        makeTestProjectCodeWithSnippet(`
+    <View style={{ ...(props.style || {}) }} data-uid='aaa'>
+      <View
+        style={{
+          backgroundColor: '#aaaaaa33',
+          width: '50%',
+          height: '20%',
+          right: 200,
+          bottom: 320,
+          top: 0,
+          left: 0
+        }}
+        data-uid='bbb'
+        data-testid='bbb'
+      />
+    </View>
+    `),
+        'await-first-dom-report',
+      )
+
+      const targetElement = EP.elementPath([
+        [BakedInStoryboardUID, 'scene-aaa', 'app-entity'],
+        ['aaa', 'bbb'],
+      ])
+
+      await selectComponentsForTest(initialEditor, [targetElement])
+
+      const viewBounds = initialEditor.renderedDOM.getByTestId('bbb').getBoundingClientRect()
+
+      const canvas = initialEditor.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+      const viewCenter = canvasPoint({
+        x: viewBounds.left + viewBounds.width / 2,
+        y: viewBounds.top + viewBounds.height / 2,
+      })
+
+      await mouseDownAtPoint(canvas, viewCenter)
+      await mouseMoveToPoint(canvas, offsetPoint(viewCenter, canvasPoint({ x: 15, y: 15 })))
+
+      await initialEditor.dispatch(
+        [CanvasActions.setUsersPreferredStrategy(ConvertToAbsoluteAndMoveStrategyID)],
+        true,
+      )
+
+      await mouseUpAtPoint(canvas, offsetPoint(viewCenter, canvasPoint({ x: 15, y: 15 })))
+
+      expect(getPrintedUiJsCode(initialEditor.getEditorState())).toEqual(
+        makeTestProjectCodeWithSnippet(
+          `<View style={{ ...(props.style || {}) }} data-uid='aaa'>
+        <View
+          style={{ backgroundColor: '#aaaaaa33', width: '50%', height: '20%', right: 185, bottom: 305, top: 15, left: 15, position: 'absolute', }}
```

Comment:
This test will break, as the width and height of the target will now be fixed: https://github.com/concrete-utopia/utopia/blob/603552b7cfd199a9c07847b7729c2d85144989f8/editor/src/components/canvas/canvas-strategies/strategies/convert-to-absolute-and-move-strategy.spec.browser2.tsx#L1174